addons/website/tools.py

```diff
@@ -72,11 +72,11 @@ def MockRequest(
         request.httprequest.url = werkzeug.urls.url_join(url_root, path)
     if website:
         request.website_routing = website.id
-    if country_code:
+    if country_code or city_name:
         try:
-            request.geoip._city_record = odoo.http.geoip2.models.City(['en'], country={'iso_code': country_code})
+            request.geoip._city_record = odoo.http.geoip2.models.City(['en'], country={'iso_code': country_code}, city={'names': {'en': city_name}})
```

Comment:
Don't have to check if `country_code` or `city_name` are set before giving them in parameter of `odoo.http.geoip2.models.City`. It seems to me to give `{'iso_code': None}` when country_code is not given (same for `city_name`). 🤔

Reply:
Maybe we could do something like:

```suggestion
            request.geoip._city_record = odoo.http.geoip2.models.City(['en'], country=country_code and {'iso_code': country_code}, city=city_name and {'names': {'en': city_name}})
```